src/main/java/appeng/services/CompassService.java

```diff
@@ -71,6 +75,22 @@ public CompassService( @Nonnull final File worldCompassFolder, @Nonnull final Th
 		return this.executor.submit( new CMDirectionRequest( coord, maxRange, cc ) );
 	}
 
+	/**
+	 * Ensure the a compass service is removed once a world gets unloaded by forge.
+	 *
+	 * @param event the event containing the unloaded world.
+	 */
+	@SubscribeEvent
+	public void unloadWorld( WorldEvent.Unload event )
+	{
+		if( Platform.isServer() )
+		{
+			final CompassReader compassReader = this.worldSet.remove( event.world );
```

Comment:
Is it 100% guaranteed, that the world is contained? else `.close()` will result in `NPE`